siliconcompiler/package/cache.py

```
@@ -0,0 +1,443 @@
+            return 0.0
+
+        with self.__lock:
+            base = self.__retry_delay
+            backoff = self.__retry_backoff
+
+        if base <= 0:
+            return 0.0
+
+        nominal = base * (backoff ** (attempts - 1))
+        jitter = PathCache.RETRY_JITTER
+        delay = random.uniform(nominal * (1 - jitter), nominal * (1 + jitter))
```

Ruff (0.16.0)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)